src/frontend/src/optimizer/optimizer_context.rs

```diff
@@ -13,11 +13,9 @@
 // limitations under the License.
 
 use core::fmt::Formatter;
-use std::cell::{Cell, RefCell, RefMut};
 use std::collections::HashMap;
-use std::marker::PhantomData;
-use std::rc::Rc;
-use std::sync::Arc;
+use std::sync::atomic::{AtomicI32, AtomicU32, AtomicUsize, Ordering};
+use std::sync::{Arc, RwLock, RwLockWriteGuard};
```

Comment:
If we use `parking_lot:RwLock` we can avoid the `.unwrap()` after every `.read()` and `.write()`.